File: backend/src/main/java/project/saloon/Saloon.java
package project.saloon;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import project.schedule.Schedule;
import project.service.Service;
import project.user.User;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Set;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "saloon")
public class Saloon implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String name;
    private String postalCode;
    private String city;
    private String country;
    private String status;  // open or closed
    private String type;    // salão cabeleireiro / barbeiro / esteticista ...
    private String contact;

    // Horario do salão
    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "schedule")
    private Schedule schedule;


    // Dono do salão
    @ManyToOne
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;


    // Serviços do salão
    @JsonIgnore
    @OneToMany(mappedBy = "saloon")
    private Set<Service> services;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public User getOwner() {
        return owner;
    }

    public void setOwner(User owner) {
        this.owner = owner;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPostalCode() {
        return postalCode;
    }

    public void setPostalCode(String postalCode) {
        this.postalCode = postalCode;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getContact() {
        return contact;
    }

    public void setContact(String contact) {
        this.contact = contact;
    }

    public Schedule getSchedule() {
        return schedule;
    }

    public void setSchedule(Schedule schedule) {
        this.schedule = schedule;
    }

    public Set<Service> getServices() {
        return services;
    }

    public void setServices(Set<Service> services) {
        this.services = services;
    }
}

File: mobile_app/lib/models/Saloon.dart
import 'package:mobile_app/models/User.dart';

class Saloon{

  int id;
  String name;
  String postalCode;
  String city;
  String country;
  String status;
  String type;
  String contact;
  String description;
  String address;
  User owner;
  String image;


  Saloon();

  Saloon.fromMap(Map<String, dynamic> data){
    id=data['id'];
    name=data['name'];
    postalCode=data['postalCode'];
    city=data['city'];
    country=data['country'];
    status=data['status'];
    type=data['type'];
    contact=data['contact'];
    description=data['description'];
    address=data['address'];
    owner=User.fromMap(data['owner']);
    image=data['image'];
  }

  Map<String, dynamic> toMap() {
    return {
      'id':id,
     'name':name,
     'postalCode':postalCode,
     'city':city,
     'country':country,
     'status':status,
     'type':type,
     'contact':contact,
     'description':description,
     'address':address,
     'owner':owner.toMap(),
     'image':image,    
    };
  }

}
